Libplanet/Net/Swarm.cs

```diff
@@ -976,6 +983,45 @@ private async Task RefreshAllocate(CancellationToken cancellationToken)
             }
         }
 
+        private async Task BroadcastTxAsync<T>(
+            BlockChain<T> blockChain,
+            TimeSpan broadcastTxInterval,
+            CancellationToken cancellationToken)
+            where T : IAction, new()
+        {
+            while (!cancellationToken.IsCancellationRequested)
+            {
+                try
+                {
+                    await Task.Delay(broadcastTxInterval, cancellationToken);
+
+                    await Task.Run(
+                        () =>
+                        {
+                            List<TxId> txIds = blockChain.Store
+                                .IterateBroadcastTransactionIds()
+                                .ToList();
```

Comment:
We need to test if non-broadcasting staged transactions are actually never broadcasted to peers.  If I've read this patch correctly the current test suite would pass even if we replace `.IterateBroadcastTransactionIds()` with `.IterateStagedTransactionIds()` here.

Reply:
I believe the commit 6b8d2e080ffac22350d4def7b56b32b2e9fe49c3 does not fix what I addressed above…  I meant our test code *should not* pass if we replace `.IterateBroadcastTransactionIds()` with `.IterateStagedTransactionIds()` here.  In other words, it's about test coverage.

Reply:
That commit you mentioned is not for this case but for using lock. I added test for this case [here](https://github.com/planetarium/libplanet/pull/297/files#diff-29be8fcde7b01328ce4fda0a59dffa1cR604). Could you please look at it?

Reply:
Ah, okay.  I'm going to take a look!

Reply:
`TxStagedNotToBroadcast()` indeed addresses the case.  :100: